play/src/pusher/models/Space.ts

```diff
@@ -202,51 +206,48 @@ export class Space implements CustomJsonReplacerInterface {
             });
     }
 
-    public notifyMe(watcher: ExSocketInterface, subMessage: SubMessage, user: SpaceUser | undefined = undefined) {
+    public notifyMe(watcher: ExSocketInterface, subMessage: SubMessage) {
         watcher.emitInBatch(subMessage);
     }
 
-    private isWatcherTargeted(
-        watcher: ExSocketInterface,
-        subMessage: SubMessage,
-        user: SpaceUser | undefined = undefined
-    ) {
+    private isWatcherTargeted(watcher: ExSocketInterface, user: SpaceUserExtended) {
         const filtersOfThisSpace = watcher.spacesFilters.filter((spaceFilters) => spaceFilters.spaceName === this.name);
```

Comment:
This seems very suspicious to me.

Why do we need to filter out the right filter on a give watcher?

Maybe we are doing this wrong. Rather than the space containing a list of watchers, maybe the space should contain a list of filters, and each filter should be connected to a watcher?

This way, you could iterate over the filters without having to iterate over the watcher first?